archivessnake/archival_objects_by_collection.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+"""Exports information about all archival objects in a resource tree.
+
+   usage: python archival_objects_by_collection.py resource_id
+"""
+
+import argparse
+import configparser
+import csv
+import json
+import os
+
+from asnake.aspace import ASpace
+
+
+class DataFetcher:
+    def __init__(self, resource_id):
+        config = configparser.ConfigParser()
+        config.read('local_settings.cfg')
+        if os.path.isfile('data.csv'):
+            raise Exception("data.csv already exists and would be overwritten. Please move or delete this file before running the script again.")
+        self.aspace = ASpace(baseurl=config.get('ArchivesSpace', 'baseURL'), username=config.get('ArchivesSpace', 'username'), password=config.get('ArchivesSpace', 'password'))
+        self.repo = self.aspace.repositories(2)
+        self.resource_id = int(resource_id)
+
+    def run(self):
+        writer = csv.writer(open('data.csv', 'w'))
+        for record in self.repo.resources(self.resource_id).tree.walk:
+            if record.jsonmodel_type == 'archival_object':
+                data = self.get_object_data(record)
+                print(data)
+                writer.writerow(data)
+
+    def get_object_data(self, obj):
+        return [obj.uri, obj.resource.title, obj.ancestors[0].title, obj.title,
```

Comment:
This (I think specifically `obj.title`) threw an error when encountering an archival object with no `title` (this is allowed if there's a date). The way I've worked around this in the past is by using `display_title`.